includes/data-stores/class-wc-customer-data-store.php

```diff
@@ -0,0 +1,242 @@
+
+		$meta_key_to_props = array(
+			'paying_customer' => 'is_paying_customer',
+			'first_name'      => 'first_name',
+			'last_name'       => 'last_name',
+		);
+
+		foreach ( $meta_key_to_props as $meta_key => $prop ) {
+			if ( ! in_array( $prop, $changed_props ) ) {
+				continue;
+			}
+
+			if ( update_user_meta( $customer->get_id(), $meta_key, $customer->{"get_$prop"}( 'edit' ) ) ) {
+				$updated_props[] = $prop;
+			}
+		}
+
+		if ( in_array( 'billing', $changed_props ) ) {
+			update_user_meta( $customer->get_id(), 'billing_first_name', $customer->get_billing_first_name( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_last_name', $customer->get_billing_last_name( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_company', $customer->get_billing_company( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_phone', $customer->get_billing_phone( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_email', $customer->get_billing_email( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_postcode', $customer->get_billing_postcode( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_city', $customer->get_billing_city( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_address_1', $customer->get_billing_address( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_address_2', $customer->get_billing_address_2( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_state', $customer->get_billing_state( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'billing_country', $customer->get_billing_country( 'edit' ) );
+		}
+
+		if ( in_array( 'shipping', $changed_props ) ) {
+			update_user_meta( $customer->get_id(), 'shipping_first_name', $customer->get_shipping_first_name( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_last_name', $customer->get_shipping_last_name( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_company', $customer->get_shipping_company( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_postcode', $customer->get_shipping_postcode( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_city', $customer->get_shipping_city( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_address_1', $customer->get_shipping_address( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_address_2', $customer->get_shipping_address_2( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_state', $customer->get_shipping_state( 'edit' ) );
+			update_user_meta( $customer->get_id(), 'shipping_country', $customer->get_shipping_country( 'edit' ) );
+		}
+	}
+
+	/**
+	 * Gets the customers last order.
+	 *
+	 * @since 2.7.0
+	 * @param WC_Customer
+	 * @return WC_Order|false
+	 */
+	public function get_last_order( &$customer ) {
+		global $wpdb;
+
+		$last_order = $wpdb->get_var( "SELECT posts.ID
```

Comment:
WPDB cannot handle arrays/implodes so you either get in a situation where you have to prepare part of it and append the part with the implode, or you can use esc_sql directly. To me esc_sql is cleaner in these situations. @justinshreve Thoughts?